modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue

```diff
@@ -221,7 +222,7 @@ export default {
         });
       }
     },
-    getSlatLabelKey () {
+    getEditRelationshipLabel () {
       if (this.field.editor === 'AposImageRelationshipEditor') {
```

Comment:
There should be no hardcoding of a specific relationship type in `AposInputRelationship`. So this label override must become part of the relationship field type configuration on the serv er side. Same for the "edit relationship" icon which is different in this design from the generic one if I recall properly.

Then we can extend my previous work in schema/index.js to allow it to come from the image piece type so that every relationship field doesn't have to specifically configure it.

Reply:
Made some modifs. Label and icon are set in the relationship config. There are passed as a fallback to field if none locally is set. if editorIcon is passed we use this as a button instead of the context menu (because for now we have no others links in the context menu.)
Let me know if it fits for you.